apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -1071,8 +1071,8 @@ defmodule Champions.Battle.Simulator do
     {target, history}
   end
 
-  defp process_executions_over_time(unit, history) do
-    Enum.reduce(unit.executions_over_time, {unit, history}, fn execution_over_time, {unit_acc, history_acc} ->
+  defp process_executions_over_time(unit_initial_state, current_unit, history) do
+    Enum.reduce(unit_initial_state.executions_over_time, {current_unit, history}, fn execution_over_time, {unit_acc, history_acc} ->
```

Comment:
We're using the initial unit's executions over time, but when we call `process_execution_over_time` we're not passing in the unit's initial state. That means the armor in the calculation gets affected by this steps' modifiers!